File: DenseArray.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);
        for (auto &x:arr)
            cin >> x;
        int ans=0;
        for (int i=0; i<n-1; i++) {
            int a = min(arr[i], arr[i+1]);
            int b = max(arr[i], arr[i+1]);
            while (a*2<b) {
                a *= 2;
                ans++;
            }
        }
        cout << ans << endl;
    }

    return 0;
}

File: Python/DenseArray.py
def main():
    t = int(input())
    while t:
        t -= 1
        n = int(input())
        arr = [int(input()) for i in range(n)]
        ans = 0
        for i in range(n-1):
            a = min(arr[i], arr[i+1])
            b = max(arr[i], arr[i+1])
            while a*2 < b:
                a *= 2
                ans += 1
        print(ans)


if __name__ == '__main__':
    main()
